### DISCUSSION TITLE
JID question

### DISCUSSION THREAD

**Discussion by debow77** [sidekiq/sidekiq#6221] [General]
Is `include Sidekiq::Worker` still required to get access to JID inside the WorkerJob or ApplicationJob?

Sidekiq 7.2.2
Redis 4.8.1

My job
```PlayJob < ApplicationJob

def perform
....
check_if_cancelled!
end
```

ApplicationJob
```
ApplicationJob < ActiveJob::Base

  def cancelled?
    # Use c.exists? on Redis >= 4.2.0
    Sidekiq.redis do |c|
      if c.respond_to? :exists?
        c.exists?("cancelled-#{jid}") == 1
      else
        c.exists("cancelled-#{jid}") == 1
      end
    end
  end
  
  def check_if_cancelled!(play = nil, pid = nil)
  ....
  status = cancelled?
  end
```

If I include `Sidekiq::Worker` in either of the above I can't submit the job.
`PlayJob.set(queue: queue_name.to_sym).perform_later(play, queue_name, project, shared_project)
`

Without Sidekiq:Worker it appears I can't leverage JID for the cancel job method.
https://github.com/sidekiq/sidekiq/wiki/FAQ#how-do-i-cancel-a-sidekiq-job

Do I need to use `Sidekiq::Client.push` instead. Anytime I add `include Sidekiq::Worker` it seems to break submitting a job.

**Comment by mperham** [comment]
FYI `Sidekiq::Worker` has been renamed to `Sidekiq::Job` now.

Yes, `jid` is the Sidekiq::Job term. ActiveJob calls it `provider_job_id`. Since you are using ActiveJob and not native Sidekiq::Jobs, you'll need to use that term instead. 

```ruby
job = PlayJob.set(...).perform_later
job.provider_job_id
```

**Reply by debow77** [reply]
Thanks, that fixed the issue. Going to  convert to using native Sidekiq::Jobs but have many workflows to change.
